ElectronClient/gui/NoteText.jsx

```diff
@@ -1662,10 +1663,13 @@ class NoteTextComponent extends React.Component {
 			});
 		}
 
+		const isViewingTrash = this.props.notesParentType === 'SmartFilter' && this.props.selectedSmartFilterId === TRASH_FILTER_ID;
+
 		if (note.markup_language === MarkupToHtml.MARKUP_LANGUAGE_MARKDOWN && editorIsVisible) {
 			toolbarItems.push({
 				tooltip: _('Bold'),
 				iconName: 'fa-bold',
+				enabled: !isViewingTrash,
```

Comment:
Are these `enabled` properties really necessary? How are the toolbar buttons and menu items handled when we are in preview-only mode? I think a function is used to toggle them.

Reply:
In preview mode the toolbar buttons are not shown (just a subset are).  For trash, I was showing them but disabled.  It was a token effort to discourage editing in trash.

The big question here is what should we allow in trash?  I was thinking no note editing.  Perhaps we should enforce preview only mode in trash?  Or a separate "trash" viewing mode where everything is disabled or hidden?

Reply:
IMO in trash we should not allow anything, but viewing the note (source and/or peview) and restoring it.
So we'd have to make the editor read-only in trash or just allow preview-only mode. I usually prefer to read the source, but this doesn't matter - it's just a personal pref.

Reply:
Ok I'll put this on the discussion list. At the moment, all that's implemented here is a "poor mans" read-only mode.

Reply:
Laurent is going to make the decisions anyway. I just expressed my opinion. A lot of people will be very happy about a trash feature.

Reply:
Thanks @tessus, opinions always welcome.  BTW it still has some significant steps to go before it can even be considered for accepting, but the concept works, at least locally.

Reply:
I agree that trashed notes should not be editable at all. Perhaps we should introduce a concept of read-only note, as that would be used for archiving notes too.

Reply:
> Perhaps we should introduce a concept of read-only note

Would you want that added in this PR, or as a separate feature PR and applied to trash later?

Am happy either way, just concerned about the size of this PR for review. Though it would mean that the notes in the trash would be somewhat editable in the interim.

Reply:
Indeed that would make the PR even more complex, so let's keep that for later. For now, if you can find a way to make them non-editable, even if it's a bit of a hack, that would be good enough. If it's too complicated, we can even leave them editable for now, it's up to you, and in another PR we can deal with making them read-only.